app/crud.py

```diff
@@ -0,0 +1,295 @@
+from . import models, schemas
+
+
+def get_steps(db: Session, skip: int = 0, limit: int = 100):
+    """Gets steps from the DB."""
+    return db.query(models.Step).offset(skip).limit(limit).all()
+
+
+def get_step_by_id(db: Session, step_id: int):
+    """Gets a single step by ID from the DB."""
+    return db.query(models.Step).filter(models.Step.id == step_id).first()
+
+
+def create_step(db: Session,
+                step: schemas.StepCreate,
+                optional_settings={}
+                ):
+    """
+    Creates a step in the DB with a lot of data processing.
+
+    Combines the optional_settings supplied in parameter with the config .yaml.
+    It combines these parameters into properly formatted values for fields.
+    """
+    # If we have optional settings, we load the config and substitute those.
+    if optional_settings:
+        settings = load_and_substitute_config(CONFIG_FILE, optional_settings)
+    else:
+        # Otherwise we just use the default config:
+        settings = config
+
+    db_step = models.Step(**step.dict())
+    db_step.status = 'prelaunch'
+    db_step.script_version = settings["version_number"]
+
+    dir_head = settings['folders']['working_dir']
+    dir_tail = '/' + settings['cc_batch']
+    further_params = "-L " + settings['runtime_configurations']['log_level']
+
+    # Some steps can run on only one process and thus have no processes param.
+    # These steps are marked by 'no_p_param: True' in the config.yaml.
+    if not settings["scripts"][db_step.step_name].pop('no_p_param', False):
+        further_params += ' -P ' \
+                          + str(config['runtime_configurations']['processes'])
+
+    # Let's go over and process the parameters:
+    for key, value in settings["scripts"][db_step.step_name].items():
+        if key == 'script_file':
+            db_step.script_file = value
+        elif key == 'input':
+            db_step.input = dir_head + value + dir_tail
+        elif key == 'output':
+            db_step.output = dir_head + value + dir_tail
+        elif key == 'hardwired_params':
+            further_params += ' ' + value
+        else:
+            # Most parameters have a simple value:
+            if isinstance(value, str):
+                further_params += " -" + key + " " + value
+            # If the parameter requires special treatment, then it is a dict.
+            elif value['is_path']:
+                # If it is a path we must append it to the project root dir.
+                # and may have to append the current batch to it.
+                if value.get('no_batch_in_path'):
+                    further_params += " -" + key + " " + dir_head + value[key]
+                else:
+                    further_params += " -" + key \
+                                      + " " + dir_head + value[key] + dir_tail
+
+    db_step.further_params = further_params
+    db.add(db_step)
+    db.commit()
+    db.refresh(db_step)

```

Comment:
I checked it and you are right, we don't need it here. 

Default behaviour of commit() is that it releases the object, so a new one will be fetched anyway, no need for refresh().

By the way: we need the reloaded object because it is returned by this method and rendered as a response.